### DISCUSSION TITLE
Is it possible to compile a static site from sitepress-rails?

### DISCUSSION THREAD

**Discussion by benjaminoakes** [sitepress/sitepress#59] [Q&A]
I've come across `sitepress compile` to build a static site.  Can that be used within `sitepress-rails` in some way?

**Comment by bradgessler** [comment]
Yep! You could hook up a rake task that compiles the site into the Rails `./public` directory that runs before (or after) the `assets:precompile` command. Rails would then serve up those pages from the `./public/` directory, depending on how your web server is configured.

**Reply by benjaminoakes** [reply]
Thanks @bradgessler 😄

Is there a prebuilt Rake task or would one need to be written from scratch? If from scratch, do you have any pointers on where to start?

**Comment by bradgessler** [comment]
I'd start from the CLI code at https://github.com/sitepress/sitepress/blob/main/sitepress-cli/lib/sitepress/cli.rb#L50-L82

You could either shell out to that or tie the Compiler class directly into the Rakefile.
